src/errors.ts

```diff
@@ -0,0 +1,343 @@
+import cwdRelativePath from "./cwd-relative-path";
+import {ErrorCode} from "./error-codes";
+import normalizeSeparators from "./normalize-separators";
+import rootRelativePath from "./root-relative-path";
+import {ErrorDetails, Options} from "./types";
+
+/**
+ * Create an error indicating that the file could not be parsed.
+ *
+ * @param file The path of the file that could not be parsed.
+ * @param detail The reason why the file could not be parsed.
+ * @returns The error details.
+ */
+export const couldNotParse = (file: string, detail: string): ErrorDetails => ({
+    markerID: null,
+    code: ErrorCode.couldNotParse,
+    reason: `Could not parse ${file}: ${detail}`,
+});
+
+/**
+ * Create an error indicating that a different comment style is being used.
+ *
+ * @param markerID The ID of the marker that is affected.
+ * @param line The line number where the marker is located.
+ * @returns The error details.
+ */
+export const differentCommentSyntax = (markerID: string, line: number) => ({
+    markerID,
+    reason: `Sync-start tags for '${markerID}' given in different comment styles. Please use the same style for all sync-start tags that have identical identifiers.`,
+    location: {line},
```

Comment:
I might be misunderstanding the error, but should there be multiple locations reported here? If the issue is that two different uses of a marker have different comment styles.

Reply:
Currently, checksync can only report on one line per error, not any previous ones. This error is for when the comment style of the tag at this position is different to the comment style of the first tag with the same marker ID. The error output will include a link to the file and line of the offending tag, and the user can scroll to investigate further from there.